dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -124,15 +132,16 @@ function Catalog(props: ICatalogProps) {
   const allRepos = uniq(charts.map(c => c.attributes.repo.name));
   const allProviders = uniq(csvs.map(c => c.spec.provider.name));
   const allCategories = uniq(
-    charts
-      .map(c => categoryToReadable(c.attributes.category))
+    categories
+      .map(c => categoryToReadable(c.name))
       .concat(flatten(csvs.map(c => getOperatorCategories(c)))),
   ).sort();
 
   useEffect(() => {
     fetchCharts(cluster, namespace, repo);
     getCSVs(cluster, namespace);
-  }, [cluster, namespace, repo, fetchCharts, getCSVs]);
+    fetchChartCategories(cluster, namespace, currentRepo);
```

Comment:
I don't think we should remove categories if a repository is selected (to only show categories of that repository). It would be a weird UX. Some examples:

https://artifacthub.io/packages/search?page=1&kind=3
https://operatorhub.io/?provider=%5B%22Alibaba+Cloud%22%5D

You can see that all the categories are still shown even if the current filters don't contain all of them.

Reply:
Mmm, I get your point but I don't any strong opinion about that. 
On the one hand, leaving the whole list of categories would result in a more consistent view, since users won't see any disappearing elements.
But..., on the other hand, considering users having their repos without categories, when they suddenly add, let's say, bitnami repo, a list of categories will appear and they won't be able to get rid of them.

We are returning the number of charts in each category, perhaps we can display this number to prevent users from clicking on non-existent categories.

I don't know, I'm open to both options as I mentioned. What's your opinion, @absoludity ?

Reply:
Let me put it the other way. If you first select a category (e.g. "CMS") and then select a repo that doesn't contain that category (e.g. "my-repo"), this would remove the category from list. 

In that scenario, the user cannot deselect the category, so they would be stuck with an empty list of apps. If to fix that you automatically remove the category filter, the whole list of applications will be shown but the user is still expecting CMS apps because they have not interacted with that filter again.

(hope I explained myself)

Reply:
Thank you for the clarification. However, I think I'm missing something beacause I believe the scenario you mention will still work. Here is a gif with the current state (ignore the catalog empty message):

Even if the CMS categories continue being selected, since it follows an OR logic, as soon as the user clicks on an existing category, charts will appear.

![filtersCategories](https://user-images.githubusercontent.com/11535726/104642457-36940a00-56ab-11eb-9f57-409dc59c3b56.gif)

Reply:
Thanks for the gif.

> Even if the CMS categories continue being selected, since it follows an OR logic, as soon as the user clicks on an existing category, charts will appear.

and importantly, the currently active filters (`Category: CMS` and `Repository: harbor`) are still displayed above and so can still be deselected individually.

> I don't know, I'm open to both options as I mentioned. What's your opinion, @absoludity ?

Same - I don't feel strongly either way and so, as always in that circumstance, I'd defer to people who do. Although I do think it'd be a simpler UX to only see the categories that are relevant (given that, as above, there is no issue deselecting active filters which are no longer relevant) there are others who argue that people will be looking to the place where they turned a filter on to be able to turn it off again and so recommend having *both* visible (see Be redundant when displaying applied filters at in this [Filtering best practises article](https://uxdesign.cc/crafting-a-kickass-filtering-ux-beea1798d64b)). So I'd go with Andres' recommendation (unless you feel less strongly Andres when you see that you can deselect them trivially).

Reply:
Thank you for your comments and for sharing the article; from this point of view, it kinda makes sense not to hide unuseful categories.
Moreover, I've just discovered another reason for going this way: these blue top level filters (`filter-summary`) are **not** accessible: keyboard navigation is not possible, so people with visual problems won't be able to remove the filters from this section. Hence, this redundancy is not being guaranteed for everybody, wherefore I also agree with the initial Andrés' recommendation.